Add search param in `getWPThemesPage` method and migrate endpoint to v2.0

pmusolino opened the pull request:
### Description

This PR can be tested through https://github.com/wordpress-mobile/WordPress-iOS/pull/24262

[WordPressKit.zip](https://github.com/user-attachments/files/19339848/WordPressKit.zip)

This is part of the work needed to fix this issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/23153

In WPiOS, the theme search was not functioning correctly because the search was conducted locally. As a result, since it also supports pagination, the search only considered the elements that had been fetched. If some elements were beyond the already downloaded data, they were not displayed.

In this PR, I migrated the endpoint `rest/v1.2/themes` to v2.0, as v1.2 has been deprecated. I also introduced the `search` parameter, which allows users to search for themes by keyword. Please refer to the linear issue for more details `CMM-55/cannot-find-seedlet-theme`.


---

- [ ] Please check here if your pull request includes additional test coverage.
- [ ] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

pmusolino committed:
Add search parameter to `getWPThemesPage` method

pmusolino committed:
update: Package.swift

dangermattic commented:
<!--
  0 Errors
  1 Warning: <code>Package.swift</code> was...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

pmusolino committed:
Revert "update: Package.swift"

This reverts commit a5a49e7bc1b3d15a30ae81254b2ccc504e77e848.

pmusolino committed:
Revert "Add search parameter to `getWPThemesPage` method"

This reverts commit 074fc43d7b404acbe0b481c6d6e3d9e46b162025.

pmusolino committed:
Add search parameter to theme retrieval methods

pmusolino committed:
update: Change WordPressKit binary target URL and checksum

dangermattic commented:
<!--
  0 Errors
  1 Warning: <code>Package.swift</code> was...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

pmusolino committed:
Remove unused filter parameter from ThemeServiceRemote
- Delete `ThemeRequestFilterKey` constant in ThemeServiceRemote.m
- Remove `search` parameter assertion in `getThemesForBlogId` method
- Eliminate filter parameter from `getThemesForBlogId` API call

pmusolino committed:
update `Package.swift`

pmusolino committed:
Change WordPressComRESTAPIVersion from 1_2 to 2_0 in ThemeServiceRemote, `getWPThemesPage` with search method

pmusolino committed:
update package.swift

pmusolino committed:
Merge branch 'wpios-edition' into add-search-param-in-theme-service-remote

pmusolino committed:
update: Package.swift

pmusolino committed:
update: ThemeServiceRemoteTests

pmusolino committed:
update: Package.swift

pmusolino committed:
update: ThemeServiceRemote header file

pmusolino committed:
update: package.swift

pmusolino committed:
fix: endpoint version in method `testThatGetThemesWorks`

pmusolino committed:
update: Package.swift

pmusolino committed:
fix: handle optional search parameter in getWPThemesPage method

pmusolino committed:
update: Package.swift

kean commented:
LGTM – the changes to the package file look good. I will approve it once there is a respective PR in WordPress-iOS with these changes where they could be tested.

pmusolino commented:
Opened the respective PR in WordPress-iOS repo https://github.com/wordpress-mobile/WordPress-iOS/pull/24262

kean reviewed (approved):
This looks good. Please feel free to incorporating it in `wpios-edition`, but make sure to pull the changes first so your binary have all changes from both `wpios-edition` and your branch.
